test/run/test_stages.py

```diff
@@ -521,3 +521,23 @@ def test_ovf(self):
             assert capacity == str(json.loads(res.stdout)["virtual-size"])
             pop_size = ovf_tree_disk.attrib["{http://schemas.dmtf.org/ovf/envelope/1}populatedSize"]
             assert pop_size == str(os.stat(vmdk).st_size)
+
+
+    def test_dnf4_mark(self):
+        datadir = self.locate_test_data()
+        testdir = os.path.join(datadir, "stages", "dnf4.mark")
+
+        with self.osbuild as osb, tempfile.TemporaryDirectory(dir="/var/tmp") as outdir:
+            osb.compile_file(os.path.join(testdir, "tree.json"), exports=["tree"], output_dir=outdir)
+
+            tree = os.path.join(outdir, "tree")
+            assert os.path.isdir(tree)
+
+            # we're going to verify that packages in the tree are marked according to
+            r = subprocess.run(["dnf", "--installroot", tree, "repoquery", "--installed" "--qf", "%{name},%{reason}\n"],
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE,
+                               encoding="utf8",
+                               check=True)
+
+            assert "dnf,user" in r.stdout
```

Comment:
Let's also check that the rest of the packages are not marked `user`.  That way we would catch a bug that, for example, marked everything user instead of just dnf.

Reply:
Maybe something like:
```python
            for line in r.stdout.split("\n"):
                if line.startswith("dnf"):
                    assert "dnf,user" in line
                else:
                    assert not line.endswith(",user")
```

so that we don't have to figure out what every package should be marked us, just that nothing else is marked `user`.